### ISSUE TITLE
[Bug] EntityFramework Navigation Property randomly not include the collection data

### ISSUE THREAD

**Issue by hermanthotan** [dotnet/efcore#14350]
I have encounter an issue during perform firstordefault without 
> query = query.OrderBy(x => Guid.NewGuid());

> 

It able to retrieve the navigation property, but I would like to get the random question. After I added the above line, the navigation property it sometimes return some data and sometimes not return at all.

```
var query = _questionRepository.Table;           
query.Include(x => x.Options)
query = query.OrderBy(x => Guid.NewGuid());
return query.FirstOrDefault();
```

Can someone investigate this.

EF Core 2.2.0

**Comment by hermanthotan** [comment]
@smitpatel What is the conclusion, how to fix this issue? a bit lose when reading the comment

**Comment by smitpatel** [comment]
The issue is in current Include Pipeline which we intend to fix in 3.0 release (tentatively). So that is when above query would work as is in EF Core.

In the meanwhile you can write query differently to work-around the issue.
Looking at query, you want to select 1 random record and load it's collection navigation.
Work-arounds:
1. Just get the record from database and then use either explicit load/lazy load to load the navigation property. It is either way 2 queries with include so this one wouldn't have much perf impact.
2. Select the record first and then do manual join to get the both entities in the result set, then apply client projection to change the shape. This will bring both the entities in context so tracking would fix up the navigation.
